### DISCUSSION TITLE
Forwarding requests for one domain to an internal nameserver?

### DISCUSSION THREAD

**Discussion by johanneskastl** [0xERR0R/blocky#459] [Q&A] [answered]
Hi all,

just getting started with Blocky and really like it this far.

One thing I might have misunderstood in the docs:

> Conditional forwarding to external DNS server

I first understood this to mean that I can tell blocky to forward all requests for `example.org` to an internal nameserver, while handling all other requests with the default name servers.

But I think this is the feature where I can use different upstream servers depending on which "group" I am sorted into, or which subnet the request is coming from.

**Question**:
Is it possible to tell blocky to use (a) different nameserver(s) for requests to one domain like `example.org`?

Thanks in advance!
Johannes

**Comment by 0xERR0R** [comment, marked as answer]
> **Question**: Is it possible to tell blocky to use (a) different nameserver(s) for requests to one domain like `example.org`?

Yes, it is possible with conditional forwarding. Following example:
```yaml
upstream:
  default:
    - 1.1.1.1
conditional:
    mapping:
        fritz.box: 192.168.178.1
```

Query "example.com" will use "1.1.1.1" as upstream DNS server. Query for "fritz.box" or "subdomain.fritz.box" will be forwarded to "192.168.178.1"

**Reply by johanneskastl** [reply]
Thanks for the fast reply.
And yes it is working fine, I just tested this.

One thing I find a little unfortunate is that this expects a string containing one or multiple nameservers, while the `upstream` section uses a yaml list.
But that is just nitpicking. It is clearly stated in the documentation (now that I know where to look :-) ) and is working as expected.

**Reply by 0xERR0R** [reply]
> One thing I find a little unfortunate is that this expects a string containing one or multiple nameservers, while the `upstream` section uses a yaml list. But that is just nitpicking. It is clearly stated in the documentation (now that I know where to look :-) ) and is working as expected.

It has a "historical background": initially, it was possible to configure only one server. Later I introduced multiple servers and I wanted to keep configuration format backward compatible. Changing it to yaml list would break existing configuration.
